### ISSUE TITLE
Composite Key for `idAccessor`

### ISSUE THREAD

**Issue by zhermin** [icflorescu/mantine-datatable#315]
**Describe the bug**
My table needs unique keys and I've found out about `idAccessor`. However, my unique key is currently a composite key and the accessor can only accept a string of the field name(s). Not sure if it's possible currently but if it is, how do I set it such that it's something like `{"id1-id2"}`? Thanks a lot for all the hard work! 

**Expected behavior**
`idAccessor` can be separated by a delimiter/separator so that multiple accessors/fields can be concatenated to form a unique key. Example: `idAccessor={"id1|id2"}`.

**Comment by icflorescu** [comment]
That's an interesting idea. I haven't thought about composite IDs so far.
Maybe we could implement this with a function instead of using a string with a separator?
I'm thinking we could change the `idAccessor` prop to be of type `string | (record: T) => unknown`.

This way, you could do something like this:
```ts
<DataTable
  // ...
  idAccessor={({ attr1, attr2 }) => `${attr1}:${attr2}`}
/>
```

What do you think?

**Comment by zhermin** [comment]
Great idea! I think allowing users to optionally set it using a function (on top of the current string implementation) is a good midpoint since composite keys are likely not a common problem for most cases, so no need to break the current approach. But don't worry about this too much if it's non-trivial, I can find workarounds in the meantime. 

Thanks for the quick response as well! Really love the work you and your community have put into this project :)

**Comment by icflorescu** [comment]
This doesn't break the current approach.
Just published `v2.5.6`, in which `idAccessor` can be a `string` **or** a function:
```ts
idAccessor?: string | ((record: T) => Key);
```
The `Key` type comes from `React` and means `string | number` if I remember correctly.

Thank you for the kind words and for using Mantine DataTable!
